internal/operators/lvm/lvm_operator.go

```diff
@@ -147,56 +171,109 @@ func (o *operator) GetMonitoredOperator() *models.MonitoredOperator {
 }
 
 // GetHostRequirements provides operator's requirements towards the host
-func (o *operator) GetHostRequirements(ctx context.Context, cluster *common.Cluster, _ *models.Host) (*models.ClusterHostRequirementsDetails, error) {
+func (o *operator) GetHostRequirements(ctx context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirementsDetails, error) {
 	log := logutil.FromContext(ctx, o.log)
 	preflightRequirements, err := o.GetPreflightRequirements(ctx, cluster)
 	if err != nil {
-		log.WithError(err).Errorf("Cannot retrieve preflight requirements for cluster %s", cluster.ID)
+		log.WithError(err).Errorf("Cannot retrieve preflight requirements for host %s", host.ID)
 		return nil, err
 	}
+
+	if len(cluster.Hosts) <= 3 {
+		return preflightRequirements.Requirements.Master.Quantitative, nil
+	}
+
+	role := common.GetEffectiveRole(host)
+
+	if role == models.HostRoleWorker && *cluster.HighAvailabilityMode == models.ClusterHighAvailabilityModeFull {
+		return preflightRequirements.Requirements.Worker.Quantitative, nil
+	}
+
+	// blocking auto assign for LVM
+	if role == models.HostRoleAutoAssign {
+		err := fmt.Errorf("Cannot Get host Requirment for auto assign role for host %s", host.ID)
+		log.WithError(err)
+		return nil, err
+	}
+
+	// default to Master
 	return preflightRequirements.Requirements.Master.Quantitative, nil
 }
 
 // GetPreflightRequirements returns operator hardware requirements that can be determined with cluster data only
 func (o *operator) GetPreflightRequirements(context context.Context, cluster *common.Cluster) (*models.OperatorHardwareRequirements, error) {
+	// unsupportted before 4.11
+	if ok, _ := common.BaseVersionLessThan(openshiftVersion4_11, cluster.OpenshiftVersion); ok {
+		return &models.OperatorHardwareRequirements{}, fmt.Errorf("LVM is not supported")
+	}
+
 	dependecies, err := o.GetDependencies(cluster)
 	if err != nil {
 		return &models.OperatorHardwareRequirements{}, err
 	}
-	return &models.OperatorHardwareRequirements{
+
+	baseHardwareRequirement, _ := GetBaseRequirements(GetPage)
```

Comment:
Fetching from the web?  How will this work when the hosts don't have internet access?  Or GitHub is down?